Move theming support from Beta to Main

jordanjones243 opened the pull request:
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #63 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)


## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

<!-- Generated by sourcery-ai[bot]: start summary -->

## Summary by Sourcery

Move theming support for the popover component from beta to main, enabling customization through design tokens. Refactor the component to replace hardcoded size and color values with design tokens, and update documentation to guide users on restyling the component.

New Features:
- Introduce theming support for the popover component, allowing customization through design tokens.

Enhancements:
- Refactor the popover component to use design tokens for size and color values, replacing hardcoded values.

Documentation:
- Add documentation on how to restyle the popover component using design tokens.

<!-- Generated by sourcery-ai[bot]: end summary -->

jordanjones243 committed:
feat: refactor color token structure with tier 3 tokens #63

jordanjones243 committed:
perf: replace hardcoded size values with design tokens

jordanjones243 committed:
build: sync package-lock.json

jordanjones243 committed:
refactor(lint): clean up errors and use new config

semantic-release-bot committed:
chore(release): 3.3.0-beta.1 [skip ci]

# [3.3.0-beta.1](https://github.com/AlaskaAirlines/auro-popover/compare/v3.2.3...v3.3.0-beta.1) (2024-07-01)

### Features

* refactor color token structure with tier 3 tokens [#63](https://github.com/AlaskaAirlines/auro-popover/issues/63) ([e5a59fd](https://github.com/AlaskaAirlines/auro-popover/commit/e5a59fd54c186cd47c0d591a50424db20dad3e99))

### Performance Improvements

* replace hardcoded size values with design tokens ([2c3c68d](https://github.com/AlaskaAirlines/auro-popover/commit/2c3c68db7ff11ddc37a5200119c4d13c61d7ad80))

jordanjones243 committed:
fix(popover): update styles to reconfigure trigger and box-shadow

semantic-release-bot committed:
chore(release): 3.3.0-beta.2 [skip ci]

# [3.3.0-beta.2](https://github.com/AlaskaAirlines/auro-popover/compare/v3.3.0-beta.1...v3.3.0-beta.2) (2024-08-09)

### Bug Fixes

* **popover:** update styles to reconfigure trigger and box-shadow ([b771780](https://github.com/AlaskaAirlines/auro-popover/commit/b771780250a03d749422fe8236981aa85b4bb365))

jordanjones243 committed:
perf: update deps and add name as attribute when custom registered

semantic-release-bot committed:
chore(release): 3.3.0-beta.3 [skip ci]

# [3.3.0-beta.3](https://github.com/AlaskaAirlines/auro-popover/compare/v3.3.0-beta.2...v3.3.0-beta.3) (2024-08-15)

### Performance Improvements

* update deps and add name as attribute when custom registered ([e5ceef4](https://github.com/AlaskaAirlines/auro-popover/commit/e5ceef4ca9bebe8f49e2355fd7736ab7deba0e78))

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/AlaskaAirlines/auro-popover?pullRequest=77) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/AlaskaAirlines/auro-popover?pullRequest=77) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: jordanjones243<br/>:x: semantic-release-bot<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/AlaskaAirlines/auro-popover?pullRequest=77) it.</sub>

jordanjones243 committed:
fix: resolve minor color theming issues

BREAKING CHANGE: trigger major release for color theme support #63

jordanjones243 committed:
perf: update dependencies

jordanjones243 committed:
perf: refactor custom component registration config

semantic-release-bot committed:
chore(release): 4.0.0-beta.1 [skip ci]

# [4.0.0-beta.1](https://github.com/AlaskaAirlines/auro-popover/compare/v3.3.0-beta.3...v4.0.0-beta.1) (2024-09-30)

### Bug Fixes

* resolve minor color theming issues ([89b344a](https://github.com/AlaskaAirlines/auro-popover/commit/89b344a05937df074cd260edfa67cfb0f29e4934))

### Performance Improvements

* refactor custom component registration config ([d080c60](https://github.com/AlaskaAirlines/auro-popover/commit/d080c609faec0c6bd990204f5d72ef5c35c4ad8b))
* update dependencies ([74ab81a](https://github.com/AlaskaAirlines/auro-popover/commit/74ab81a5881baf1efce5cdd37a3a34b6e05c3847))

### BREAKING CHANGES

* trigger major release for color theme support #63

blackfalcon commented:
:tada: This PR is included in version 4.0.0-beta.1 :tada:

The release is available on:
- [npm package (@beta dist-tag)](https://www.npmjs.com/package/@aurodesignsystem/auro-popover/v/4.0.0-beta.1)
- [GitHub release](https://github.com/AlaskaAirlines/auro-popover/releases/tag/v4.0.0-beta.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:

jordanjones243 committed:
docs: remove beta releases from CHANGELOG

jason-capsule42 committed:
docs: fix typo

blackfalcon commented:
:tada: This PR is included in version 4.0.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@aurodesignsystem/auro-popover/v/4.0.0)
- [GitHub release](https://github.com/AlaskaAirlines/auro-popover/releases/tag/v4.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
